tst/CTA.Rules.Test/Actions/ActionHelpers/ConfigMigrateTests.cs

```diff
@@ -11,6 +11,84 @@ namespace CTA.Rules.Test.Actions.ActionHelpers
+    <add key = ""UseCustomizationData"" value = ""false""/>
+  </appSettings >
+
+  <system.web>
+    <compilation debug = ""true"" targetFramework = ""4.7.2"" />
+    <httpRuntime targetFramework = ""4.6.1"" requestValidationMode = ""2.0"" />
+    <sessionState mode = ""InProc"" />
+    <pages>
+      <namespaces>
+        <add namespace=""System.Web.Optimization"" />
+      </namespaces>
+
+    <controls>
+      <add assembly = ""Microsoft.AspNet.Web.Optimization.WebForms"" namespace=""Microsoft.AspNet.Web.Optimization.WebForms"" tagPrefix=""webopt"" />
+    </controls></pages>
+    <httpModules>
+      <add name = ""ContainerDisposal"" type=""Autofac.Integration.Web.ContainerDisposalModule, Autofac.Integration.Web"" />
+      <add name = ""PropertyInjection"" type=""Autofac.Integration.Web.Forms.PropertyInjectionModule, Autofac.Integration.Web"" />
+      <add name = ""TelemetryCorrelationHttpModule"" type=""Microsoft.AspNet.TelemetryCorrelation.TelemetryCorrelationHttpModule, Microsoft.AspNet.TelemetryCorrelation"" />
+      <add name = ""ApplicationInsightsWebTracking"" type=""Microsoft.ApplicationInsights.Web.ApplicationInsightsHttpModule, Microsoft.AI.Web"" />
+    </httpModules>
+    <globalization uiCulture = ""en-US"" culture=""en-US"" />
+  </system.web>
+  <system.codedom>
+    <compilers>
+      <compiler language = ""c#;cs;csharp"" extension="".cs"" type=""Microsoft.CodeDom.Providers.DotNetCompilerPlatform.CSharpCodeProvider, Microsoft.CodeDom.Providers.DotNetCompilerPlatform, Version=2.0.1.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35"" warningLevel=""4"" compilerOptions=""/langversion:default /nowarn:1659;1699;1701"" />
+      <compiler language = ""vb;vbs;visualbasic;vbscript"" extension="".vb"" type=""Microsoft.CodeDom.Providers.DotNetCompilerPlatform.VBCodeProvider, Microsoft.CodeDom.Providers.DotNetCompilerPlatform, Version=2.0.1.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35"" warningLevel=""4"" compilerOptions=""/langversion:default /nowarn:41008 /define:_MYTYPE=\&quot;Web\&quot; /optionInfer+"" />
+    </compilers>
+  </system.codedom>
+</configuration > ");
+
+            File.WriteAllText("web.config", webConfig);
+
+            var templateContent = @"{}";
+            var outputDir = "";
+
+            var configuration = (Configuration)loadWebConfigMethod.Invoke(configMigrateInstance, new object[] { outputDir });
+
+            // Invoke method and read contents of method output
+            var content = (JObject)processWebConfigMethod.Invoke(configMigrateInstance, new object[] { configuration, templateContent });
+
+            var methodParams = new object[] { content, outputDir };
+            addAppSettingsJsonFileMethod.Invoke(configMigrateInstance, methodParams);
+
+            var appSettingsContent = File.ReadAllText(Path.Combine(outputDir, "appsettings.json"));
+            File.Delete(Path.Combine(outputDir, "appsettings.json"));
+            File.Delete(Path.Combine(outputDir, "web.config"));
```

Comment:
If modifying the file system in unit tests, use a `[OneTimeSetUp]` function to create a directory containing all test files and write files there. Then create a `[OneTimeTearDown]` function to delete the directory you created. This is a safe way to ensure that test files are always cleaned up (OneTimeTearDown is always called at the end of tests, even if an error occurs during a test). 

Note: The test class will need to be decorated with a `[TestFixture]` attribute as well in order for `[OneTime*]` attributes to take effect. I believe there are other examples of this in the project available for reference as well.

Reply:
I was following the structure of the rest of the tests in the file, that test was only added for me to see how ConfigMigrate worked. I can remove it if necessary though

Reply:
If this test is not confirming correctness, then we can delete it. If it is, then it should be ok to leave in

Reply:
You can keep it if it adds code coverage for tests, if it was purely for you to debug just keep it locally without checking it in on the branch for your own benefit. If you think however it does provide  useful unit test coverage please keep it but implement it as Jon mentioned with a OnetimeTearDown attribute to ensure these always get cleaned up. \cta\tst\CTA.Rules.Test\SetupTests.cs has an example of this if you want to use it.